api/app/signals/apps/api/ml_tool/utils.py

```diff
@@ -11,8 +11,8 @@ def get_category_from_prediction(category_url):
     if not hasattr(resolved, 'kwargs'):
         return category_url, False
 
-    slug = resolved.kwargs['sub_slug'] if 'sub_slug' in resolved.kwargs else resolved.kwargs['slug']
-    parent_isnull = False if 'sub_slug' in resolved.kwargs else True
+    slug = resolved.kwargs['slug'] if 'slug' in resolved.kwargs else resolved.kwargs['parent_lookup_parent__slug']
```

Comment:
Make this `slug = resolved.kwargs['slug']`, slug is present both for main and sub categories. If it weren't this code would do the wrong thing (`parent_lookup_parent__slug` is only present for a sub category see line below --- we cannot use it to get the main category slug).

Reply:
I see what you are saying. And it seems redundant, will replace this by `slug = resolved.kwargs['slug']`